### ISSUE TITLE
Difficulties in VarDict-1.5.4 installation

### ISSUE THREAD

**Issue by macko-tipor** [AstraZeneca-NGS/VarDict#74]
Hi,

I tried to upgrade VarDict-1.5.1 to VarDict-1.5.4 to take advantage of options -u and -UN.
But I have some difficulties with VarDict-1.5.4 installation.

I followed the commands included in the Readme.md document.

git clone --recursive https://github.com/AstraZeneca-NGS/VarDictJava.git
cd ~/VarDictJava
./gradlew clean installDist

The following error message was shown:

        Downloading https://services.gradle.org/distributions/gradle-4.8-bin.zip
        Exception in thread "main" java.net.ConnectException: Connection timed out (Connection timed out)
	        at java.net.PlainSocketImpl.socketConnect(Native Method)
	        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	        at java.net.Socket.connect(Socket.java:589)
	        at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:673)
	        at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
	        at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
	        at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	        at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	        at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	        at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	        at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1156)
	        at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050)
	        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1564)
	        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
	        at org.gradle.wrapper.Download.downloadInternal(Download.java:58)
	        at org.gradle.wrapper.Download.download(Download.java:44)
	        at org.gradle.wrapper.Install$1.call(Install.java:61)
	        at org.gradle.wrapper.Install$1.call(Install.java:48)
	        at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:65)
	        at org.gradle.wrapper.Install.createDist(Install.java:48)
	        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:128)
	        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)

I got the same message when trying:
./gradlew distZip

Then I installed gradle-4.9, added the gradle-4.9/bin folder to the PATH and repeated the ./gradlew clean installDist command.
The same error message was shown again.

Therefore, I tried a different way.
I downloaded VarDict-1.5.4.zip, created an install folder in VarDictJava folder, and unzipped VarDict-1.5.4.zip there. The bin and lib folders were created.
I have successfully used VarDict-1.5.1 in the same way so far.
However, when I tried to use VarDict-1.5.4, the following error message came:

        java.util.concurrent.ExecutionException: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	        at com.astrazeneca.vardict.VarDict.somaticParallel(VarDict.java:394)
	        at com.astrazeneca.vardict.VarDict.nonAmpVardict(VarDict.java:301)
	        at com.astrazeneca.vardict.VarDict.start(VarDict.java:80)
	        at com.astrazeneca.vardict.Main.run(Main.java:159)
	        at com.astrazeneca.vardict.Main.main(Main.java:31)
        Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	        at java.lang.String.charAt(String.java:658)
	        at com.astrazeneca.vardict.VarDict.isHasAndNotEquals(VarDict.java:4422)
	        at com.astrazeneca.vardict.VarDict.modifyCigar(VarDict.java:6825)
	        at com.astrazeneca.vardict.VarDict.parseSAM(VarDict.java:1673)
	        at com.astrazeneca.vardict.VarDict.toVars(VarDict.java:2917)
	        at com.astrazeneca.vardict.VarDict$ToVarsWorker.call(VarDict.java:5753)
	        at com.astrazeneca.vardict.VarDict$SomdictWorker.call(VarDict.java:5774)
	        at com.astrazeneca.vardict.VarDict$SomdictWorker.call(VarDict.java:5758)
	        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	        at java.lang.Thread.run(Thread.java:748)

I repeated the installation steps with VarDict-1.5.3 and VarDict-1.5.2.
I got the same error message when trying to run VarDict-1.5.3.
On the contrary, VarDict-1.5.2 works well.

Now I can apply VarDict-1.5.2 which includes the -u and -UN options.
But what can I do to install the latest version?

My java version is 1.8.0_171.

Thank you very much for your help!

**Comment by mjafin** [comment]
Hi @macko-tipor,
Could you try the ready built jars from https://github.com/AstraZeneca-NGS/VarDictJava/releases ?

**Comment by macko-tipor** [comment]
Hi mjafin,

I have tried them. I downloaded and unzipped the three latest binaries.

VarDict-1.5.2 works well, so I can now exploit the -u and -UN options.
The depth bias problem has been solved this way. Everything is clear now in that respect.

But the latest two binaries both give the exception "String index out of range: -1".
What can be the problem? Maybe our system settings has something incompatible with them?
I would like to follow the new VarDict versions in the future and I am afraid of getting the same error message with each one if this problem remains unsolved.
